src/GTP.cpp

```diff
@@ -77,6 +77,7 @@ void GTP::setup_default_parameters() {
 #endif
     cfg_puct = 0.85f;
     cfg_softmax_temp = 1.0f;
+    // Keep in sync with UCTSearch::should_resign
```

Comment:
I don't like this, easy to mess up. Making cfg_resignpct = -1 be something like "use default logic", and then putting 10 as a default in that logic, is also not perfect but seems preferable.

Reply:
Too bad std::optional is C++17. But boost has a version too, and it's header only, so that would be usable.

Reply:
Logic now lives in UCTSearch